x-pack/test/api_integration/services/usage_api.ts

```diff
@@ -5,57 +5,60 @@
  * 2.0.
  */
 
-import { UsageStatsPayload } from '@kbn/telemetry-collection-manager-plugin/server';
+import type { UsageStatsPayload } from '@kbn/telemetry-collection-manager-plugin/server';
 import {
   ELASTIC_HTTP_VERSION_HEADER,
   X_ELASTIC_INTERNAL_ORIGIN_REQUEST,
 } from '@kbn/core-http-common';
-import { FtrProviderContext } from '../ftr_provider_context';
+import type { FtrProviderContext } from '../ftr_provider_context';
 
 export interface UsageStatsPayloadTestFriendly extends UsageStatsPayload {
   // Overwriting the `object` type to a more test-friendly type
   stack_stats: Record<string, any>;
 }
 
+export interface GetTelemetryStatsOpts {
+  authHeader: Record<string, string>;
+}
+
 export function UsageAPIProvider({ getService }: FtrProviderContext) {
   const supertest = getService('supertest');
+  const supertestWithoutAuth = getService('supertestWithoutAuth');
 
   async function getTelemetryStats(
     payload: {
       unencrypted: true;
       refreshCache?: boolean;
     },
-    options?: { authHeader: Record<string, string> }
+    opts?: GetTelemetryStatsOpts
   ): Promise<Array<{ clusterUuid: string; stats: UsageStatsPayloadTestFriendly }>>;
   async function getTelemetryStats(
     payload: {
       unencrypted: false;
       refreshCache?: boolean;
     },
-    options?: { authHeader: Record<string, string> }
-  ): Promise<Array<{ clusterUuid: string; stats: string }>>;
-  async function getTelemetryStats(
-    payload: {
-      unencrypted: false;
-      refreshCache?: boolean;
-    },
-    options?: { authHeader: Record<string, string> }
+    opts?: GetTelemetryStatsOpts
   ): Promise<Array<{ clusterUuid: string; stats: string }>>;
   async function getTelemetryStats(
     payload: {
       unencrypted?: boolean;
       refreshCache?: boolean;
     },
-    options?: { authHeader: Record<string, string> }
+    opts?: GetTelemetryStatsOpts
   ): Promise<Array<{ clusterUuid: string; stats: UsageStatsPayloadTestFriendly | string }>> {
-    const { body } = await supertest
+    const client = opts?.authHeader ? supertestWithoutAuth : supertest;
```

Comment:
Explicitly using different clients because sometimes overriding the pre-authenticated `supertest` can be a source of flakiness.